processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -757,38 +772,48 @@ public int add(String value)
     }
 
     @Override
-    public int getSortedId(String value)
+    public DimLookup sort()
     {
-      return delegate.getSortedId(Strings.nullToEmpty(value));
+      return new NullValueConverterDimLookup(delegate.sort());
```

Comment:
i think it is nice to preserve the behavior here that nulls are returned as empty strings

in general in Druid, nulls and empty strings are treated the same